### DISCUSSION TITLE
Install on different path on Windows?

### DISCUSSION THREAD

**Discussion by RauchenwaldC** [Woolverine94/biniou#40] [Q&A]
Is there a way to install (at least the models) on a different path?
My primary drive, which stores the user profile folder, does not have enough space, and I'd like to ideally install everything but at least the models on a secondary (in my case, 4TB SSD) drive, but I can't find an option to do so.

**Comment by Woolverine94** [comment]
Hello @RauchenwaldC,

It must be feasible on both Windows and Linux, but the modifications of the installers are not the same :

  - For linux : launch the installation process from the folder where you want it to be installed
  - For windows, it's more complicated, and i'll try to commit a fix ASAP using a variable instead of a hard-coded path 

Also note that the cache folder is not completely handled by biniou, so some components (mostly the dependencies of the used models) will install inside this folder, which is usually on the system drive. 

Prerequisites software will -of course- always install on the system drive.

**Reply by RauchenwaldC** [reply]
Thanks for the quick response and the work you are putting into this.

**Comment by Woolverine94** [comment]
@RauchenwaldC,

Commit 8d2537b allow Windows users to define a custom path for biniou directory, when installing with `install_win.cmd` . 

Proceed as follow :

  - Download and edit install_win.cmd
  - Modify `set DEFAULT_BINIOU_DIR="%userprofile%"` to `set DEFAULT_BINIOU_DIR="E:\datas\ai"` (for example). 
  - Only use absolute path (e.g.: `E:\datas\somedir` and not `.\datas\somedir`)
  - Don't had a trailing slash (e.g.: `E:\datas\somedir` and not `E:\datas\somedir\` )
  - Don't add a "biniou" suffix to your path (e.g.: `E:\datas\somedir\biniou`), as the biniou directory will be created by the git clone command 
  - Save and launch install_win.cmd
  - Enjoy !
